src/broadcast/participant.rs

```diff
@@ -325,22 +318,16 @@ impl BroadcastParticipant {
         info!("Generating round two broadcast messages.");
 
         let data = BroadcastData::from_message(message)?;
-        let data_bytes = serialize!(&data)?;
+        //let data_bytes = serialize!(&data)?;
         // todo: handle this more memory-efficiently
         let mut others_minus_leader = self.other_participant_ids.clone();
         others_minus_leader.retain(|&id| id != leader);
-        let messages: Vec<Message> = others_minus_leader
-            .iter()
-            .map(|&other_participant_id| {
-                Message::new(
-                    MessageType::Broadcast(BroadcastMessageType::Redisperse),
-                    message.id(),
-                    self.id,
-                    other_participant_id,
-                    &data_bytes,
-                )
-            })
-            .collect();
+        let messages = self.message_for_other_participants(
+            MessageType::Broadcast(BroadcastMessageType::Redisperse),
+            message.id(),
+            self.id,
+            data,
+        )?;
```

Comment:
The bug is here. Notice before we were iterating over all entries in `others_minus_leader` which is just `other_participant_ids` minus the `leader` ID. Your current version iterates over everything _including_ the leader.

I would leave your current code as is, that generates messages for all participants, but after generating the messages, call the `retain` function that filters out the leader. Just how the old code does. Lmk if you have any questions about this.